unix-domain-socket/src/main/scala/akka/stream/alpakka/unixdomainsocket/javadsl/UnixDomainSocket.scala

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.unixdomainsocket.javadsl
+
+import java.io.File
+import java.util.Optional
+import java.util.concurrent.CompletionStage
+
+import scala.compat.java8.OptionConverters._
+import scala.compat.java8.FutureConverters._
+import akka.NotUsed
+import akka.actor.{ActorSystem, ExtendedActorSystem, Extension, ExtensionId, ExtensionIdProvider}
+import akka.stream.alpakka.unixdomainsocket.scaladsl.{UnixDomainSocket => ScalaUnixDomainSocket}
+import akka.stream.javadsl.{Flow, Source}
+import akka.stream.Materializer
+import akka.util.ByteString
+import jnr.unixsocket.UnixSocketAddress
+
+import scala.concurrent.duration.Duration
+
+object UnixDomainSocket extends ExtensionId[UnixDomainSocket] with ExtensionIdProvider {
+
+  /**
+   * Represents a prospective UnixDomainSocket server binding.
+   */
+  class ServerBinding private[akka] (delegate: ScalaUnixDomainSocket.ServerBinding) {
```

Comment:
Could you make this and other classes final?

Reply:
Sure.